dotnet/CoreLib/Search/SearchClient.cs

```diff
@@ -143,7 +143,12 @@ await foreach ((MemoryRecord memory, double relevance) in matches.WithCancellati
         return result;
     }
 
-    public async Task<MemoryAnswer> AskAsync(string index, string question, MemoryFilter? filter = null, CancellationToken cancellationToken = default)
+    public async Task<MemoryAnswer> AskAsync(
+        string index,
+        string question,
+        double minRelevanceScore = 0,
```

Comment:
I think we need a different approach to support the min score and a list of filters.
something like

```
class MemoryFilter :  : TagCollection {
   ..like MemoryFilter now, without the relevance value...
}

// new class, not sure about the name
class MultiMemoryFilter {

    float MinRelevance

    IList<MemoryFilter> Filters

    // we could move "limit" here too, but perhaps on a different PR

}
```

Reply:
Should this be refactored in a separate PR? Feel like the method overloads, configuration, etc could all get revamped as a dedicated change.